cli/lsp/config.rs

```diff
@@ -15,19 +15,57 @@ pub struct ClientCapabilities {
   pub workspace_did_change_watched_files: bool,
 }
 
+#[derive(Debug, Clone, Default, Deserialize)]
```

Comment:
Not necessary to fix in this PR, but generally we should avoid "Clone" unless necessary. I suppose this one needs to be Clone because WorkspaceSettings is clone. But do we really make copies of WorkspaceSettings? Maybe - but that seems like it could result in some situations where there are two different workplace configs.

Reply:
I will try to remove `Clone` from `WorkspaceSettings`, I can't remember why it was there.

Reply:
Ok, it was there, because `Config` was cloned, but it appears that this was a legacy hold over from when we first adapted `rust_analyzer` structure, where sometimes the config was cloned.  There was only one place where it was cloned, which it actually shouldn't have been.  Thanks for pointing it out.